File: backend/backend/measurement.py
import json
import glob

from django.core.exceptions import ValidationError
from django.http import HttpResponse
from projects.models import Measurement


def get_measurement(request, measurement_id):
    try:
        m = Measurement.objects.get(id=measurement_id)
        response = {
            "description": m.description,
            "analysis": m.analysis,
            "evaluation": m.evaluation,
            "screenshot_path": m.screenshot_path,
            "setup_path": m.setup_path,
            "raw_data_path": m.raw_data_path,
            "date_time": str(m.date_time),
            "temperature": m.temperature
        }
        return HttpResponse(json.dumps(response))
    except ValidationError:
        return HttpResponse("")


def update_measurement(request, measurement_id):
    req_data = request.read().decode('utf-8')
    data = json.loads(req_data)

    try:
        m = Measurement.objects.get(id=measurement_id)
        m.description = data["description"]
        m.analysis = data["analysis"]
        m.evaluation = data["evaluation"]
        m.save()
    except ValidationError:
        return HttpResponse("Failed")

    return HttpResponse("Success")

def get_screenshot(request, measurement_id):
    req_data = request.read().decode('utf-8')
    screenshot_path = ""

    try:
        m = Measurement.objects.get(id=measurement_id)
        screenshot_path = m.screenshot_path
    except ValidationError:
        return HttpResponse("Screenshot not found")

    screenshots = glob.glob("." + screenshot_path + "/*")

    for img in screenshots:
        print(img)
        with open(img, "rb") as imageFile:
            return HttpResponse(imageFile.read(), content_type="image/png")

    return HttpResponse("No screenshots found.")

File: web-frontend/src/pages/Measurement.js
import React, {useState} from "react";
import {useParams} from "react-router-dom";
import Navigation from "../components/navigation";
import {backendUrl} from "../config";

function Measurement() {

    let {projectId, measurementId} = useParams();

    let [description, setDescription] = useState("");
    let [analysis, setAnalysis] = useState("");
    let [evaluation, setEvaluation] = useState("");
    let [screenshotPath, setScreenshotPath] = useState("");
    let [setupPath, setSetupPath] = useState("");
    let [rawDataPath, setRawDataPath] = useState("");
    let [dateTime, setDateTime] = useState("");
    let [temperature, setTemperature] = useState("");

    let [syncStatus, setSyncStatus] = useState("");

    let [changeDescription, setChangeDescription] = useState("");
    let [changeExperiment, setChangeExperiment] = useState("");
    let [changeEvaluation, setChangeEvaluation] = useState("");

    let getMeasurement = () => fetch(backendUrl + "/measurement/" + measurementId + "/get-meas")
        .then(response => response.json())
        .then(json => {
            setDescription(json.description);
            setAnalysis(json.analysis);
            setEvaluation(json.evaluation);
            setScreenshotPath(json.screenshot_path);
            setSetupPath(json.setup_path);
            setRawDataPath(json.raw_data_path);
            setDateTime(json.date_time);
            setTemperature(json.temperature);
        });

    getMeasurement();

    let updateMeasurement = (description, experiment, evaluation) => {
        let body = {
            "description": changeDescription,
            "analysis": changeExperiment,
            "evaluation": changeEvaluation
        }
        try {
            fetch(backendUrl + "/measurement/" + measurementId + "/update-meas/",
                {
                    method: "POST",
                    body: JSON.stringify(body)
                }).then(response => {
                setSyncStatus("Success")
            });
        } catch (e) {
            setSyncStatus("Failed")
        }

    }

    return (
        <div>
            <header>
                <Navigation/>
            </header>
            <div>
                Description: {description} <br/>
                Experiment: {analysis} <br/>
                Evaluation: {evaluation} <br/>
                Screenshot Path: {screenshotPath} <br/>
                Setup Path: {setupPath} <br/>
                Raw Data Path: {rawDataPath} <br/>
                Date Time: {dateTime} <br/>
                Temperature: {temperature}
            </div>
            <div>
                <div>
                    {syncStatus}
                </div>
                <h4>Edit Measurement Info</h4>
                <input
                    placeholder="Description"
                    onChange={(ev) => setChangeDescription(ev.target.value)}
                /> <br/>
                <input
                    placeholder="Experiment"
                    onChange={(ev) => setChangeExperiment(ev.target.value)}
                /> <br/>
                <input
                    placeholder="Evaluation"
                    onChange={(ev) => setChangeEvaluation(ev.target.value)}
                /> <br/>
                <button onClick={(ev) => updateMeasurement()}>Send</button>
            </div>
        </div>

    );

}

export default Measurement;